asreview/datasets.py

```diff
@@ -29,203 +29,195 @@ class DataSetNotFoundError(Exception):
-            d.dataset_id for d in self.datasets
-        ]
-        return pretty_format(dataset_dict)
-
-    def __len__(self):
-        return len(self.datasets)
+        return f"<BaseDataGroup group_id='{self.group_id}'>"
 
-    def find(self, dataset_name):
-        if dataset_name == self.dataset_id:
-            return self
-
-        all_dataset_names = [(d, d.aliases) for d in self.datasets]
-        for dataset, aliases in all_dataset_names:
-            if dataset_name.lower() in aliases:
-                return dataset
-
-        raise DataSetNotFoundError(
-            f"Dataset {dataset_name} not found"
-        )
+    def __dict__(self):
+        return {d.dataset_id: d for d in self.datasets}
 
-    def get(self, i_version=-1):
-        return self.datasets[i_version].get()
-
-    def list(self, latest_only=True):
-        if latest_only:
-            return [self.datasets[-1]]
-        return [self]
-
-
-class BaseDataGroup():
-    def __init__(self, *args):
-        """Group of datasets."""
-        self._data_sets = [a for a in args]
+    def append(self, dataset):
+        """Append dataset to group.
 
-    def __str__(self):
-        return "".join([
-            f"*******  {str(data.dataset_id)}  *******\n"
-            f"{str(data)}\n\n" for data in self._data_sets
-        ])
+        dataset: asreview.datasets.BaseDataSet
+            A asreview BaseDataSet-like object.
+        """
+        if not issubclass(dataset, BaseDataSet):
+            raise ValueError(
+                "Expected BaseDataSet or subclass of BaseDataSet."
+            )
+        self.datasets.append(dataset)
 
-    def to_dict(self):
-        return {data.dataset_id: data for data in self._data_sets}
+    def find(self, dataset_id):
+        """Find dataset in the group.
 
-    def append(self, dataset):
-        self._data_sets.append(dataset)
+        Parameters
+        ----------
+        dataset_id: str
+            Identifier of the dataset to look for. It can also be one
+            of the aliases. Case insensitive.
 
-    def find(self, dataset_name):
+        Returns
+        -------
+        asreview.datasets.BaseDataSet:
+            Returns base dataset with the given dataset_id.
+        """
         results = []
-        for d in self._data_sets:
-            try:
-                dataset_result = d.find(dataset_name)
-                results.append(dataset_result)
-            except DataSetNotFoundError:
-                pass
+        for d in self.datasets:
+            if dataset_id.lower() == d.dataset_id.lower() or \
+                    dataset_id.lower() in [a.lower() for a in d.aliases]:
+                results.append(d)
+
         if len(results) > 1:
             raise ValueError(
                 f"Broken dataset group '{self.group_id}' containing multiple"
-                f" datasets with the same name/alias '{dataset_name}'.")
+                f" datasets with the same name/alias '{dataset_id}'.")
```

Comment:
Shouldn't this check be an assertion at the initialisation of the object? A private "check for duplicates" function that is called during init and when appending.

Something like 
```python
    def _assert_unique_dataset_ids(self):
        for dataset in self.datasets:
            assert dataset.dataset_id not in self.list()
```

Reply:
When a list, append and find function are available, then shouldn't `self.datasets` be a private attribute?